Source/player.cpp

```diff
@@ -1561,56 +1561,20 @@ void __fastcall FixPlrWalkTags(int pnum)
 
 void __fastcall RemovePlrFromMap(int pnum)
 {
-	int v1; // esi
-	signed int v2; // edi
-	signed int v3; // edx
-	signed int v4; // ebx
-	char v5; // al
-	signed int v6; // edx
-	_BYTE *v7; // eax
-	signed int v8; // edi
-	int v9; // ecx
-	int v10; // [esp+Ch] [ebp-4h]
-
-	v1 = -1 - pnum;
-	v10 = pnum + 1;
-	v2 = 1;
-	do
-	{
-		v3 = v2;
-		v4 = 111;
-		do
-		{
-			if ( dPlayer[0][v3 + 111] == v1 || dPlayer[0][v3] == v1 )
-			{
-				v5 = dFlags[1][v3];
-				if ( v5 & 0x20 )
-					dFlags[1][v3] = v5 & ~DFLAG_PLAYER;
-			}
-			v3 += 112;
-			--v4;
-		}
-		while ( v4 );
-		++v2;
-	}
-	while ( v2 < 112 );
-	v6 = 0;
-	do
-	{
-		v7 = (unsigned char *)dPlayer + v6;
-		v8 = 112;
-		do
-		{
-			v9 = (char)*v7;
-			if ( v9 == v10 || v9 == v1 )
-				*v7 = 0;
-			v7 += 112;
-			--v8;
-		}
-		while ( v8 );
-		++v6;
-	}
-	while ( v6 < 112 );
+	int x, y;
+	int pNext = pnum + 1;
+	int pNextI = -(pnum + 1);
+
+	for(y = 1; y < MAXDUNY; y++)
+		for (x = 1; x < MAXDUNX; x++)
+			if ( dPlayer[x+1][y-1] == pNextI || dPlayer[x][y] == pNextI )

```

Comment:
Are you sure its `x+1` and not `x-1` here? Otherwise this is a buffer overflow.

Reply:
movsx ecx, byte ptr [eax+0x6F]
the only alternative is dPlayer[x][y+111] but that's just the same thing again.

I know the decompiler sometimes try to access naboring paramteres by just applying an offset to and array or regular pointer, but it seams to small here to actually make enough of a difference?

Reply:
Well, accessing the `dPlayer` array makes sense in the context of the function. And, if it's binary identical, then indeed this is equivalent to the original code; in which case. Add `TODO: BUGFIX, buffer overflow.` ^^